crates/core_arch/src/x86/avx2.rs

```diff
@@ -3124,7 +3137,25 @@ pub unsafe fn _mm256_srlv_epi64(a: __m256i, count: __m256i) -> __m256i {
     transmute(psrlvq256(a.as_i64x4(), count.as_i64x4()))
 }
 
-// TODO _mm256_stream_load_si256 (__m256i const* mem_addr)
+/// Load 256-bits of integer data from memory into dst using a non-temporal memory hint. mem_addr
+/// must be aligned on a 32-byte boundary or a general-protection exception may be generated. To
+/// minimize caching, the data is flagged as non-temporal (unlikely to be used again soon)
+///
+/// [Intel's documentation](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm256_stream_load_si256)
```

Comment:
This (and all other AVX2 non-temporal operations) should get the same safety comment that the older non-temporal stores have. See e.g. [here](https://doc.rust-lang.org/nightly/core/arch/x86/fn._mm_stream_si128.html).

Reply:
I checked and only non-temporal *stores* have special memory orderings on x86. x86 non-temporal loads work just like normal loads.

Reply:
@Amanieu told that that doesn't apply to streaming loads, only streaming stores.

Reply:
Oh, I didn't realize non-temporal loads even are a thing. More nightmare waiting to happen, I guess...